### ISSUE TITLE
Method OCP.wait_for_resource() can't be used to wait for anything but status

### ISSUE THREAD

**Issue by mbukatov** [red-hat-storage/ocs-ci#792]
Method OCP.wait_for_resource() can't be used to wait for anything else but status. So one can't wait for a Deployment to reach particular number of replicas running.

**Comment by gklein** [comment]
I see that wait_for_resource() can now accept a "resource_count". Does it solves this requirement?

https://github.com/red-hat-storage/ocs-ci/blob/20639878c4b081fda2a5260d924781afb6c978bd/ocs_ci/ocs/ocp.py#L364

**Comment by mbukatov** [comment]
As long as someone haven't fixed this on purpose, I don't think so. Option `resource_count` was there when I reported this issue.

**Comment by gklein** [comment]
> As long as someone haven't fixed this on purpose, I don't think so. Option `resource_count` was there when I reported this issue.

Ok. Could you please help clarify the use case for this request? I think an example could help us understand if we can somehow resolve it with the existing method or we need an extra logic here.

**Comment by mbukatov** [comment]
The current code works only with pods for small positive use cases.

Let's say you have a pod running:

```
$ oc get pod -n namespace-test-19a047a82c85406c968323448e0b4de5
NAME        READY   STATUS    RESTARTS   AGE
fio-dl8tp   1/1     Running   0          57s
```

With the current code, you can wait for this one pod via:

```
ocp_pod = ocp.OCP(kind="Pod", namespace=project.namespace)              
ocp_pod.wait_for_resource(                                              
    resource_count=1,                                                   
    condition=constants.STATUS_COMPLETED,                               
    timeout=write_timeout,                                              
    sleep=30)
```

If you deploy 3 pods, and need to wait for all 3 to be running, you can tweak `resource_count` accordingly.

But you can wait for value of `STATUS` column only. You can't wait for a pod based on value of `RESTARTS` column, which limits the negative cases one can automate. But it also means that one can't work with k8s objects/kinds which doesn't have `STATUS` column, such as Jobs or Deployments. See eg.:

```
$ oc get job -n namespace-test-19a047a82c85406c968323448e0b4de5
NAME   COMPLETIONS   DURATION   AGE
fio    0/2           6m55s      6m55s
```

What if I wanted to wait for completion of 1st pod of this job (basically waiting for value `1/2` in `COMPLETIONS` column?

**Comment by mbukatov** [comment]
The other thing is that the code would likely break when used with non pod objects, because of problems like https://github.com/red-hat-storage/ocs-ci/issues/612

**Comment by gklein** [comment]
> The current code works only with pods for small positive use cases.
> 
> Let's say you have a pod running:
> 
> ```
> $ oc get pod -n namespace-test-19a047a82c85406c968323448e0b4de5
> NAME        READY   STATUS    RESTARTS   AGE
> fio-dl8tp   1/1     Running   0          57s
> ```
> 
> With the current code, you can wait for this one pod via:
> 
> ```
> ocp_pod = ocp.OCP(kind="Pod", namespace=project.namespace)              
> ocp_pod.wait_for_resource(                                              
>     resource_count=1,                                                   
>     condition=constants.STATUS_COMPLETED,                               
>     timeout=write_timeout,                                              
>     sleep=30)
> ```
> 
> If you deploy 3 pods, and need to wait for all 3 to be ~running~ completed, you can tweak `resource_count` accordingly.
> 
> But you can wait for value of `STATUS` column only. You can't wait for a pod based on value of `RESTARTS` column, which limits the negative cases one can automate. But it also means that one can't work with k8s objects/kinds which doesn't have `STATUS` column, such as Jobs or Deployments. See eg.:
> 
> ```
> $ oc get job -n namespace-test-19a047a82c85406c968323448e0b4de5
> NAME   COMPLETIONS   DURATION   AGE
> fio    0/2           6m55s      6m55s
> ```
> 
> What if I wanted to wait for completion of 1st pod of this job (basically waiting for value `1/2` in `COMPLETIONS` column?

Based on the use case described, it make sense for me to add more flexibility to wait_for_resource() and support a custom column name instead of the hardcoded 'STATUS' column.

Here is my suggestion on how to extend it:

https://github.com/red-hat-storage/ocs-ci/pull/1077

**Comment by mbukatov** [comment]
Fixed by #1077
